geostreams/test_client.py

```diff
@@ -0,0 +1,44 @@
+from unittest import TestCase
+import numpy as np
+import redis
+
+from .client import (read_from_redis, Redis, numpy_redis_mapping,
+                     redis_set)
+
+
+def test_numpyredis():
+    r = redis.StrictRedis()
+
+    x = np.random.rand(100, 100)
+
+    redis_set(r, "x", x)
+    y = read_from_redis(r, "x")
+
+    np.testing.assert_equal(x, y)
+
+
+class ZictTests(TestCase):
```

Comment:
Let's try to avoid using the `unittest` class structure here. Pytest will do a dict equal assertion so I don't think you need `self.assertDictEqual(d["c"], di)`.

Reply:
More than happy to get rid of the class structure. How does dict assertion in pytest work? Does `assert a_dict==another_dict` work magically somehow?

Reply:
yep: https://docs.pytest.org/en/latest/assert.html#making-use-of-context-sensitive-comparisons

Reply:
definitely a case of RTFM. Basically the seconds sentence of the pytest docs is
> Due to pytest‘s detailed assertion introspection, only plain assert statements are used. See Getting Started for more examples.